src/commcare_cloud/commands/migrations/copy_files.py

```diff
@@ -253,19 +268,71 @@ def teardown_auth(plan, environment, ansible_context):
 
 def _run_auth_playbook(plan, environment, ansible_context, action):
     auth_pairs = set()
-    for target_host, source_configs in plan.items():
+    for target_host, source_configs in plan.configs.items():
         auth_pairs.update({
             (target_host, config.source_host, config.source_user) for config in source_configs
         })
 
     for target_host, source_host, source_user in auth_pairs:
-        run_ansible_playbook(
-            environment, 'setup_bidirectional_auth.yml', ansible_context, skip_check=True,
-            unknown_args=[
-                '-e', 'hostA={}'.format(target_host),
-                '-e', 'hostB={}'.format(source_host),
-                '-e', 'usernameA=root',
-                '-e', 'usernameB={}'.format(source_user),
-                '-e', 'action={}'.format(action),
-                '-e', 'bidirectional=False',
-            ])
+        if action == 'add':
+            _genearate_and_fetch_key(environment, target_host, 'root', ansible_context)
+            _set_auth_key(plan.source_env, source_host, source_user, ansible_context)
+        elif action == 'remove':
+            _set_auth_key(plan.source_env, source_host, source_user, ansible_context)
+            os.remove(ID_RSA_TMP)
+
+
+def _genearate_and_fetch_key(env, host, user, ansible_context):
+    user_args = "name={} generate_ssh_key=yes".format(user)
+    run_ansible_module(env, ansible_context, host, 'user', user_args, True, None, None)
+
+    user_home_output = PrivilegedCommand(
+        'ansible',
+        env.get_ansible_user_password(),
+        "getent passwd {} | cut -d: -f6".format(user)
+    ).run_parallel_command(host)
+    user_home = user_home_output[host]
+
+    fetch_args = "src={user_home}/.ssh/id_rsa.pub dest={key_tmp} flat=yes fail_on_missing=yes".format(
+        user_home=user_home, key_tmp=ID_RSA_TMP
+    )
+    run_ansible_module(env, ansible_context, host, 'fetch', fetch_args, True, None, None)
+
+
+def _set_auth_key(env, host, user, ansible_context, remove=False):
+    state = 'absent' if remove else 'present'
+    args = "user={} state={} key={{{{ lookup('file', '{}') }}}}".format(user, state, ID_RSA_TMP)
+    run_ansible_module(env, ansible_context, host, 'authorized_key', args, True, None, None)
+
+
+class PrivilegedCommand():
```

Comment:
will replace this with the one from https://github.com/dimagi/commcare-cloud/pull/1972 once it's merged